src/libraries/System.Private.CoreLib/src/System/Decimal.cs

```diff
@@ -1585,12 +1588,51 @@ public static decimal CreateTruncating<TOther>(TOther value)
             }
         }
 
+        /// <inheritdoc cref="INumberBase{TSelf}.IsCanonical(TSelf)" />
+        public static bool IsCanonical(decimal value)
+        {
+            uint scale = (byte)(value._flags >> ScaleShift);
+
+            if (scale == 0)
+            {
+                // We have an exact integer represented with no trailing zero
+                return true;
+            }
+
+            // We have some value where some fractional part is specified. So,
+            // if the least significant digit is 0, then we are not canonical
+
+            if (value._hi32 == 0)
+            {
+                return (value._lo64 % 10) != 0;
+            }
+
+            var significand = new UInt128(value._hi32, value._lo64);
+            return (significand % 10U) != 0U;
+        }
+
+        /// <inheritdoc cref="INumberBase{TSelf}.IsComplexNumber(TSelf)" />
+        static bool INumberBase<decimal>.IsComplexNumber(decimal value) => false;
+
+        /// <inheritdoc cref="INumberBase{TSelf}.IsEvenInteger(TSelf)" />
+        public static bool IsEvenInteger(decimal value)
+        {
+            decimal truncatedValue = Truncate(value);
+            return (value == truncatedValue) && ((truncatedValue._lo64 & 1) == 0);
+        }
```

Comment:
It's not quite the same for `decimal` because we need to compare against the `truncatedValue` so we can handle `1m`, `1.0m`, `1.0000000000000000000000000000m`, and all the "in-between" variations.

The latter ones are represented as `1` is `significand: 1, scale: 0`. While `1.0` is `significand: 10, scale: 1`. This means we must "canonicalize" the value first.